inc/cobs.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+
+#include <cstdint>
+#include <cstddef>
+
+/*
+Consistent Overhead Byte Stuffing (COBS) removes a specific value from a
+list of values. The removed value can then, for example, be used as a
+frame marker.
+
+Refer to: https://en.wikipedia.org/wiki/Consistent_Overhead_Byte_Stuffing
+
+COBS considerations:
+
+1. Configurable marker byte value. The current implementation assumes
```

Comment:
How exactly does this work if the marker byte is not 0? The 0 implementation is easy to understand because you simply use overhead and altered bytes that point to the next zero location. What if the marker byte is 0x03 and you need to point to a location 3 bytes away?